### PR TITLE
Convert custom-domain-login to accessible modal.

### PR DESCRIPTION
**Pull request by jgerigmeyer**
https://trello.com/c/3B5KWFIU

red panda

### CODE REVIEW

**Comment by jgerigmeyer** [comment]
Assigned to @wlonk for code review, @stacyk and @SondraE for functional/style review:

screenshot 2018-09-12 14 38 11

File: src/js/components/header/customDomainModal.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    window.location.assign(`${baseUrl}?custom_domain=${val}`);
+  };
+
+  handleChange = (event: SyntheticInputEvent<HTMLInputElement>) => {
+    this.setState({ url: event.target.value });
+  };
+
+  render(): React.Node {
+    const footer = [
+      <Button key="cancel" label="Cancel" onClick={this.handleClose} />,
```
Comment (wlonk) [inline src/js/components/header/customDomainModal.js]: Do we handle keyboard events? Can you dismiss this with Esc?

Comment (jgerigmeyer) [reply src/js/components/header/customDomainModal.js]: Yes, that's built in to the `Modal` component. We handle it via the `onRequestClose` option: https://github.com/SFDO-Tooling/metadeploy/pull/38/files#diff-87307d3c06e1ffb42c427237c023440eR52

Comment (wlonk) [reply src/js/components/header/customDomainModal.js]: OK, cool.

**Pull request merged**